src/CodeGenerator/src/Generator/InputGenerator.php

```diff
@@ -374,9 +376,46 @@ private function inputClassRequestGetters(StructureShape $inputShape, ClassType
             }
         }
 
-        $requestUri = $requestUri ?? '';
-        $requestUri .= 'return "' . str_replace(['{', '+}', '}'], ['{$uri[\'', '}', '\']}'], $operation->getHttpRequestUri()) . '";';
+        $body['uri'] = $requestUri ?? '';
+        $body['uri'] .= '$uriString = "' . str_replace(['{', '+}', '}'], ['{$uri[\'', '}', '\']}'], $operation->getHttpRequestUri()) . '";';
+
+        $method = \var_export($operation->getHttpMethod(), true);
+
+        $class->addMethod('request')->setComment('@internal')->setReturnType(Request::class)->setBody(<<<PHP
+
+// Prepare headers
+{$body['header']}
+
+// Prepare query
+{$body['querystring']}
+
+// Prepare URI
+{$body['uri']}
+
+// Return the Request
+return new Request($method, \$uriString, \$query, \$headers, StreamFactory::create($bodyCall));
+PHP
+);
+    }
+
+    /**
+     * Convert variable to a string.
+     */
+    private function stringifyHeaderValue(string $variable, Member $member): string
+    {
+        $type = $member->getShape()->getType();
+        if ('timestamp' === $type) {
+            if (in_array($member->getLocationName(), ['Expires', 'If-Modified-Since', 'If-Unmodified-Since'])) {
+                return $variable . '->format(\DateTimeInterface::RFC1123)';
+            }
+
+            return $variable . '->format(\DateTimeInterface::ATOM)';
+        }
+
+        if ('boolean' === $type) {
+            return $variable . ' ? "true" : "false"';
+        }
 
-        $class->addMethod('requestUri')->setComment('@internal')->setReturnType('string')->setBody($requestUri);
+        return $variable;
```

Comment:
I'd add a `if ('string' ...) ` else `throw new InvalidArgumentException('not yet implemented');`

Reply:
Added!